docker-compose.yml

```diff
@@ -1,133 +1,159 @@
+    image: ghcr.io/mastodon/mastodon:nightly
     env_file: .env.production
+    environment:
+      - PORT=3000
+      - BIND=127.0.0.1
+    restart: always
     command: bundle exec puma -C config/puma.rb
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          "curl -s --noproxy localhost localhost:3000/health | grep -q 'OK' || exit 1",
+        ]
     networks:
       - external_network
       - internal_network
-    healthcheck:
-      # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:3000/health || exit 1']
     ports:
       - '127.0.0.1:3000:3000'
+    volumes:
+      - ./public/system:/mastodon/public/system
     depends_on:
       - db
       - redis
-      # - es
+      - es
+
+  sidekiq:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:nightly
+    env_file: .env.production
+    restart: always
+    command: bundle exec sidekiq
+    healthcheck:
+      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+    networks:
+      - external_network
+      - internal_network
     volumes:
       - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      - es
 
   streaming:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.9
-    restart: always
+    # build:
+    #   context: ./streaming
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon-streaming:nightly
     env_file: .env.production
-    command: node ./streaming
+    environment:
+      - PORT=4000
+      - BIND=127.0.0.1
+    restart: always
+    command: node ./streaming/index.js
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          "curl -s --noproxy localhost localhost:4000/api/v1/streaming/health | grep -q 'OK' || exit 1",
+        ]
     networks:
       - external_network
       - internal_network
-    healthcheck:
-      # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:4000/api/v1/streaming/health || exit 1']
     ports:
       - '127.0.0.1:4000:4000'
     depends_on:
       - db
       - redis
 
-  sidekiq:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.9
+  db:
+    image: postgres:16-alpine
     restart: always
-    env_file: .env.production
-    command: bundle exec sidekiq
-    depends_on:
-      - db
-      - redis
+    healthcheck:
+      test: ['CMD', 'pg_isready', '-U', 'postgres']
+    shm_size: 256mb
     networks:
-      - external_network
       - internal_network
     volumes:
-      - ./public/system:/mastodon/public/system
+      - ./postgres16:/var/lib/postgresql/data
+    environment:
+      - 'POSTGRES_HOST_AUTH_METHOD=trust'
+
+  redis:
+    restart: always
+    image: redis:7-alpine
     healthcheck:
-      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+      test: ['CMD', 'redis-cli', 'ping']
+    networks:
+      - internal_network
+    volumes:
+      - ./redis:/data
+
+  es:
```

Comment:
Maybe we could use profiles (https://docs.docker.com/compose/profiles/) to allow disabling ES, and also to enable tor / privoxy below?